hooks/general_actions.py

```diff
@@ -87,9 +92,59 @@ def generate_actions(self, sg_data, actions, ui_area):
                 action_instances.append( 
                     {"name": "publish_clipboard", 
                       "params": None,
-                      "caption": "Copy path to clipboard", 
+                      "group": "Copy to clipboard",
+                      "caption": "Path on disk",
                       "description": "Copy the path associated with this publish to the clipboard."} )
 
+        if "add_to_playlist" in actions and ui_area == "details":
+            # retrieve the 10 most recently updated non-closed playlists for this project
+
+            from tank_vendor.shotgun_api3.lib.sgtimezone import LocalTimezone
+            datetime_now = datetime.datetime.now(LocalTimezone())
+
+            playlists = self.parent.shotgun.find(
+                "Playlist",
+                [
+                    ["project", "is", sg_data.get("project")],
+                    ["sg_status", "is_not", "clsd"],
+                    {
+                        "filter_operator": "any",
+                        "filters": [
+                            ["sg_date_and_time", "greater_than", datetime_now],
+                            ["sg_date_and_time", "is", None]
+                        ]
+                    }
+                ],
+                ["code", "id", "sg_date_and_time"],
+                order=[{"field_name": "updated_at", "direction": "desc"}],
+                limit=10,
+            )
+
+            # playlists this version is already part of
+            existing_playlist_ids = [x["id"] for x in (sg_data.get("playlists") or [])]
```

Comment:
nitpick: can you do `sg_data.get("playlists", [])` instead?

Reply:
yep will fix!